lib/qri/qri.go

```diff
@@ -71,6 +76,24 @@ func (m *Module) GetConfig(thread *skylark.Thread, _ *skylark.Builtin, args skyl
 	return lib.Marshal(m.ds.Transform.Config)
 }
 
+// GetDataset returns the dataset we're transforming
+func (m *Module) GetDataset(thread *skylark.Thread, _ *skylark.Builtin, args skylark.Tuple, kwargs []skylark.Tuple) (skylark.Value, error) {
+	if m.infile == nil {
+		return skylark.None, fmt.Errorf("qri.get_dataset failed: no DataFile")
+	}
+	bytes, err := ioutil.ReadAll(m.infile)
+	if err != nil {
+		return skylark.None, err
+	}
+	var object interface{}
+	err = json.Unmarshal(bytes, &object)
```

Comment:
Can't assume provided data is JSON. Instead try:
```golang
var (
  bytes []byte
  err error
  object interface{}
)

if ds.Structure.Format == dataset.JSONDataFormat {
  bytes, err = ioutil.ReadAll(m.infile)
  if err != nil {
    return skylark.None, err
  }
} else {
  r, err := dsio.NewReader(m.ds.Structure, m.infile)
  // if err != nil ...
  buf, err := dsio.NewEntryBuffer(&dataset.Structure{Format : dataset.JSONDataFormat, Schema: m.ds.Structure.Schema})
  
  err := dsio.EachEntry(r, func(_ int, ent Entry, err error) error {
    if err != nil {
      return err
    }
    return buf.WriteEntry(ent)
  })
  // if err != nil ...
  bytes = buf.Bytes()
}

err = json.Unmarshal(bytes, &object)
// ...
```
The above is super verbose. I'll open an issue...

Reply:
qri-io/dataset#127